### PR TITLE
Upmerge 16.05.2024

### PR DESCRIPTION
**Pull request by rlubos**
Pull upmerged sdk-zephyr and sdk-mcuboot repositories.

test-sdk-nrf: sdk-nrf-pr-15351
test_ble: DRGN-22017_upmerge-16-05-2024
test_rs: KRKNWK-19014-upmerge-20240516
test-sdk-sidewalk: sdk-nrf-pr-15351

### CODE REVIEW

**Comment by MarekPieta** [comment]
Spotted a BLE VS commands related issue while testing nRF Desktop. Aligned here: https://github.com/rlubos/sdk-nrf/pull/20
Waiting for reviews from codeowners

**Review by SeppoTakalo — APPROVED** [review]

File: applications/asset_tracker_v2/src/cloud/cloud_codec/lwm2m/lwm2m_codec.c
Comment (balaji-nordic) [inline applications/asset_tracker_v2/src/cloud/cloud_codec/lwm2m/lwm2m_codec.c:32]: We need an `ARG_UNUSED(offset);` statement after line 44.

File: applications/asset_tracker_v2/tests/lwm2m_codec_helpers/src/lwm2m_codec_helpers_test.c
Comment (balaji-nordic) [inline applications/asset_tracker_v2/tests/lwm2m_codec_helpers/src/lwm2m_codec_helpers_test.c:61]: Add `ARG_UNUSED(offset);`

Comment (rlubos) [reply applications/asset_tracker_v2/tests/lwm2m_codec_helpers/src/lwm2m_codec_helpers_test.c:61]: Fixed

**Review by kkasperczyk-no — APPROVED** [review]

**Review by jayteemo — APPROVED** [review]

**Review by plskeggs — APPROVED** [review]

**Review by MarkusLassila — APPROVED** [review]

**Review by jfischer-no — APPROVED** [review]

**Review by jfischer-no — APPROVED** [review]

File: subsys/net/lib/softap_wifi_provision/softap_wifi_provision.c
Comment (glarsennordic) [inline subsys/net/lib/softap_wifi_provision/softap_wifi_provision.c:801]: Argh. Sorry. I was reviewing as you merged, just noticed this.

It seems these NULLs have a lower-case L

**Review by glarsennordic — COMMENTED** [review]
Since I just found what looks like a breaking change, and this was just merged, I am submitting my review before it is complete

File: samples/wifi/sta/sample.yaml
Comment (rlubos) [reply samples/wifi/sta/sample.yaml:32]: @krish2718 Should it? I guess it'd make sense?

File: subsys/net/lib/softap_wifi_provision/softap_wifi_provision.c
Comment (rlubos) [reply subsys/net/lib/softap_wifi_provision/softap_wifi_provision.c:801]: How in the world CI didn't catch this... Will fix

File: samples/wifi/sta/sample.yaml
Comment (krish2718) [reply samples/wifi/sta/sample.yaml:32]: Ideally yes, but this was mainly a hot fix for `nRF52840`, I will check if this is applicable to other boards as well and update `modules/hostap` if its only `nRF52840` then will need to add a board specific overlay. Thanks @glarsennordic .

**Pull request merged**
